jib-core/src/main/java/com/google/cloud/tools/jib/registry/BlobPuller.java

```diff
@@ -42,18 +45,31 @@
    */
   private final OutputStream destinationOutputStream;
 
+  // TODO: Refactor into BlobPullMonitor.
+  private final Consumer<Long> sizeConsumer;
+  private final Consumer<Long> receivedByteCountConsumer;
+  private final Duration delayBetweenCallbacks = Duration.ofMillis(100);
+
   BlobPuller(
       RegistryEndpointRequestProperties registryEndpointRequestProperties,
       DescriptorDigest blobDigest,
-      OutputStream destinationOutputStream) {
+      OutputStream destinationOutputStream,
+      Consumer<Long> sizeConsumer,
+      Consumer<Long> receivedByteCountConsumer) {
     this.registryEndpointRequestProperties = registryEndpointRequestProperties;
     this.blobDigest = blobDigest;
     this.destinationOutputStream = destinationOutputStream;
```

Comment:
What if do `this.destinationOutputStream = new ListenableCountingOutputStream(...)`? Doesn't have to save `receivedByteCountConsumer` in a field.